lib/reline/line_editor.rb

```diff
@@ -1106,6 +1137,7 @@ def update(key)
   end
 
   def input_key(key)
+    save_old_buffer
```

Comment:
how about store old_buffer_of_lines to a local variable and pass as an argument to save_past_lines like this?
`save_past_lines(old_buffer_of_lines, old_line_index, old_byte_pointer)`
This will reduce narrow scope instance variable

Reply:
Just an idea, I'm not sure it works but perhaps pushing `[old_buffer_of_lines, old_line_index, old_byte_pointer]` to `@past_lines` will make the implementation simple.

Reply:
> Just an idea, I'm not sure it works but perhaps pushing [old_buffer_of_lines, old_line_index, old_byte_pointer] to @past_lines will make the implementation simple.

Thanks for sharing your idea! I tried it, but it wasn't good enough.
Currently, I process in this order: `save_old_buffer` -> `undo` -> `@past_lines.push`.

If I change the order to `@past_lines.push` -> `undo`, it seems I need to access `@past_lines[-2]` instead of `@past_lines.last` during undo, which isn't very intuitive.

Also, I haven't investigated the cause, but when I push first, consecutive deletions don't undo properly. For example, the following test case fails:

```
Failure: test_undo_with_multiline(Reline::KeyActor::Emacs::Test)
/Users/mi/ghq/github.com/ruby/reline/test/reline/helper.rb:158:in `assert_whole_lines'
/Users/mi/ghq/github.com/ruby/reline/test/reline/test_key_actor_emacs.rb:1487:in `test_undo_with_multiline'
     1484:     assert_line_index(1)
     1485:     assert_line_around_cursor('', '')
     1486:     input_keys("\C-_", false)
  => 1487:     assert_whole_lines(["1", "2", "3"])
     1488:     assert_line_index(1)
     1489:     assert_line_around_cursor('2', '')
     1490:     input_keys("\C-_", false)
<["1", "2", "3"]> expected but was
<["1", "3"]>

diff:
? ["1", "2", "3"]

```

Reply:
> how about store old_buffer_of_lines to a local variable and pass as an argument to save_past_lines like this?
save_past_lines(old_buffer_of_lines, old_line_index, old_byte_pointer)
This will reduce narrow scope instance variable

I know what you mean. However, I want to do the same thing in both `key.char == :bracketed_paste_start` and `input_key`, which is why it's set up this way. If we use local variables in `input_key`, I'd like to use local variables in `bracketed_paste_start` as well. But adding the save process to `insert_pasted_text` would broaden its responsibilities, which is concerning. 🤔